ogr/services/forgejo/release.py

```diff
@@ -0,0 +1,172 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import datetime
+from typing import Any, Optional
+
+from pyforgejo import PyforgejoApi
+
+
+class GitTag:
+    """
+    Class representing a git tag.
+
+    Attributes:
+        name (str): Name of the tag.
+        commit_sha (str): Commit hash of the tag.
+    """
+
+    def __init__(self, name: str, commit_sha: str) -> None:
+        self.name = name
+        self.commit_sha = commit_sha
+
+    def __str__(self) -> str:
+        return f"GitTag(name={self.name}, commit_sha={self.commit_sha})"
+
+
+class ForgejoGitTag(GitTag):
+    """
+    Concrete implementation of GitTag for Forgejo.
+    """
+
+    # Using the base implementation as no additional functionality is needed.
+
+
+class Release:
+    """
+    Object that represents a release.
+    """
+
+    def __init__(self, raw_release: Any, project: Any) -> None:
+        self._raw_release = raw_release
+        self.project = project
+
+
+class ForgejoRelease(Release):
+    """
+    Concrete implementation of Release for Forgejo.
+    """
+
+    @property
+    def title(self) -> str:
+        return self._raw_release.get("name", "")
+
+    @property
+    def body(self) -> str:
+        return self._raw_release.get("description", "")
+
+    @property
+    def git_tag(self) -> GitTag:
+        return ForgejoGitTag(self.tag_name, self._raw_release.get("commit_sha", ""))
+
+    @property
+    def tag_name(self) -> str:
+        return self._raw_release.get("tag_name", "")
+
+    @property
+    def url(self) -> Optional[str]:
+        return self._raw_release.get("html_url")
+
+    @property
+    def created_at(self) -> datetime.datetime:
+        created_value = self._raw_release.get("created_at")
+        if isinstance(created_value, datetime.datetime):
+            return created_value
+        return (
+            datetime.datetime.strptime(created_value, "%Y-%m-%dT%H:%M:%SZ")
+            if created_value
+            else datetime.datetime.now()
+        )
+
+    @property
+    def tarball_url(self) -> str:
+        return self._raw_release.get("tarball_url", "")
+
+    @staticmethod
+    def get(
+        project: Any,
+        identifier: Optional[int] = None,
+        name: Optional[str] = None,
+        tag_name: Optional[str] = None,
+    ) -> "Release":
+        client = PyforgejoApi(api_url=project.forge_api_url, token=project.token)
```

Comment:
instead of initialising the client multiple times, you could create a property, you can get inspired or even use [this](https://github.com/packit/ogr/blob/ac4c629529dd1869f7872c1f06b399fcf15234f9/ogr/services/forgejo/project.py#L56) and [this](https://github.com/packit/ogr/blob/ac4c629529dd1869f7872c1f06b399fcf15234f9/ogr/services/forgejo/service.py#L34)